src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -19,18 +19,72 @@
             [puppetlabs.puppetdb.zip :as zip]
             [schema.core :as s])
   (:import [honeysql.types SqlCall SqlRaw]))
+
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Plan - functions/transformations of the internal query plan
 
-(defrecord Query [projections selection source-table alias where
-                  subquery? entity call group-by limit offset order-by])
-(defrecord BinaryExpression [operator column value])
-(defrecord InArrayExpression [table column value])
-(defrecord RegexExpression [column value])
-(defrecord ArrayRegexExpression [column value])
-(defrecord NullExpression [column null?])
-(defrecord ArrayBinaryExpression [column value])
-(defrecord InExpression [column subquery])
+(def table-names #{"catalog_resources"
+                   "certnames"
+                   "edges"
+                   "environments"
+                   "fact_paths"
+                   "fact_values"
+                   "facts"
+                   "factsets"
+                   "latest_report"
+                   "reports"
+                   "resource_events"
+                   "resource_params"})
```

Comment:
This list is possibly different in PE, I don't think I'm a fan of adding this validation because it's mostly going to be something that fails whenever someone adds a new migration and they'll have to track down where the schema error is coming from.